src/components/Widget/components/Conversation/components/Messages/components/Carousel/index.js

```diff
@@ -0,0 +1,171 @@
+import React, { useRef, useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { addUserMessage, emitUserMessage } from 'actions';
+import { PROP_TYPES } from 'constants';
+import Arrow from 'assets/arrow.svg';
+
+import './styles.scss';
+
+const Carousel = (props) => {
+  const carousel = props.message.toJS();
+
+  const handleClick = (action) => {
+    if (!action || action.type !== 'postback') return;
+    const { chooseReply } = props;
+    chooseReply(action.payload, action.title);
+  };
+
+  const scrollContainer = useRef();
+  const [leftButton, setLeftButton] = useState(false);
+  const [rightButton, setRightButton] = useState(true);
+
+  const handleScroll = () => {
+    const current = scrollContainer.current;
+    if (scrollContainer.current.scrollLeft > 0) {
```

Comment:
Minor: you could use current instead of scrollContainer.current.

Reply:
ah bah oui, indeed